ci/Makefile.jenkins

```diff
@@ -140,13 +140,17 @@ endif
 mrproper:
 	make clean_all
 	git clean -f -d -x
+	$(MAKE) -f $(THIS_FILE) apply_patches
```

Comment:
Minor comment, I'm not sure this should be part of mrproper. I was always looking at mrproper as "clean everything" target, but I might be wrong.

Reply:
I think you are right... my interpretation of `mrproper` is to restore to a clean state. If we had a fork of the Rabit repository, the clean state would include a commit that has the patch. This is just a workaround that we don't have the fork and I cannot just make a commit.

IMHO this is not common but maybe justifable :)

I will leave it for now and we can figure out a better way.